src/main/java/org/jitsi/jicofo/Participant.java

```diff
@@ -748,4 +759,27 @@ public String getMucJid()
     {
         return roomMember.getContactAddress();
     }
+
+    /**
+     * Replaces the {@link ChannelAllocator}, which is currently allocating
+     * channels for this participant (if any) with the specified channel
+     * allocator (if any).
+     * @param channelAllocator the channel allocator to set, or {@code null}
+     * to clear it.
+     */
+    public void setChannelAllocator(ChannelAllocator channelAllocator)
+    {
+        synchronized (channelAllocatorSyncRoot)
+        {
+            if (this.channelAllocator != null)
+            {
+                // There is an ongoing thread allocating channels and sending
+                // an invite for this participant. Tell it to stop.
+                this.channelAllocator.cancel();
```

Comment:
We need to make sure that ChannelAllocator once canceled will not modify Participant, because it can overwrite what was written by the new ChannelAllocator (see my comment above about "why there is no check for canceled").